ee/tabby-ui/lib/experiment-flags.ts

```diff
@@ -39,18 +39,32 @@ class ExperimentFlagFactory {
   }
 
   defineHook() {
-    return (): [{ value: boolean; description: string }, () => void] => {
-      const [storageValue, setStorageValue] = useLocalStorage(
-        this.storageKey,
-        this.defaultValue
-      )
+    return (): [
+      { value: boolean; description: string; loading: boolean },
+      () => void
+    ] => {
+      const [storageValue, setStorageValue] = useState(this.defaultValue)
+      const [loading, setLoading] = useState(true)
+
+      useEffect(() => {
+        const value = localStorage.getItem(this.storageKey)
+        if (value) {
+          setStorageValue(value === 'true')
+        }
+        setLoading(false)
+      }, [])
+
       const toggleFlag = () => {
-        setStorageValue(!storageValue)
+        const newStorageValue = !storageValue
+        setStorageValue(newStorageValue)
+        localStorage.setItem(this.storageKey, String(newStorageValue))
       }
+
       return [
         {
           value: storageValue,
-          description: this.description
+          description: this.description,
+          loading
         },
         toggleFlag
       ]
```

Comment:
Modify the hook not using useLocalStorage or the next.js will got error of rendering not aligned between server and client